### PR TITLE
Surveyor vote rate

### LINKED ISSUE #487: add API to expose vote/BAT exchange rate

### PR DESCRIPTION
**Pull request by maikelmclauflin**
fixes #487

### CODE REVIEW

File: ledger/controllers/surveyor.js [hunk 1]
```diff
@@ -415,6 +417,34 @@ v2.phase2 =
     { schema: Joi.object().keys({ submissionId: Joi.string().required().description('verification submissionId') }) }
 }
 
+/*
+   GET /v1/surveyor/{surveyorId}/voterate
```
Comment (nvonpentz) [inline ledger/controllers/surveyor.js]: Did you mean `GET /v1/surveyor/voterate/{surveyorType}/{surveyorId}`?  I see that in the routes.

File: ledger/controllers/surveyor.js [hunk 2]
```diff
@@ -708,3 +739,24 @@ module.exports.initialize = async (debug, runtime) => {
...
+    const {
+      adFree
+    } = payload
+
+    const bigProbi = new BigNumber(adFree.probi)
+    const bigVotes = new BigNumber(adFree.votes)
+    const voteRateProbi = bigProbi.dividedBy(bigVotes)
+    const voteRateBat = voteRateProbi.dividedBy(1e18)
+    const voteRate = voteRateBat.toString()
+
```
Comment (nvonpentz) [inline ledger/controllers/surveyor.js]: It looks like this code is doing something similar as https://github.com/brave-intl/bat-ledger/blob/f82076649ad124e121e2d72a5d9b88d0f7ac4ef5/ledger/controllers/wallet.js#L218-L222

Do you think there's a way to consolidate this logic?

File: test/ledger/surveyor.integration.test.js [hunk 3]
```diff
@@ -47,35 +79,45 @@ test('required cohorts are added to surveyors', async (t) => {
...
-  await postgres.query(`DELETE FROM surveyor_groups WHERE id = $1::text;`, [id])
 
-  function querySurveyor (method, surveyorId) {
-    return surveyors[method]({
+  await surveyors.remove({
+    surveyorId
+  })
+  await postgres.query(`DELETE FROM surveyor_groups WHERE id = $1::text;`, [surveyorId])
+
+  function get () {
```
Comment (nvonpentz) [inline test/ledger/surveyor.integration.test.js]: I'm worried this function name could be a bit ambiguous.  Is there a more descriptive name we could use?

File: test/ledger/surveyor.integration.test.js [hunk 4]
```diff
@@ -47,35 +79,45 @@ test('required cohorts are added to surveyors', async (t) => {
...
   } = await createSurveyor()
-  const { surveyorId: id } = publicSurveyor
+  const { surveyorId } = publicSurveyor
 
   const ledger = await connectToDb('ledger')
   const surveyors = ledger.collection('surveyors', () => {})
-  const privateSurveyor = await querySurveyor('findOne', id)
+  const privateSurveyor = await get()
 
+  const encoded = encodeURIComponent(surveyorId)
```
Comment (nvonpentz) [inline test/ledger/surveyor.integration.test.js]: nit: `encodedSurveyorId`

File: test/ledger/surveyor.integration.test.js (hunk 3 above)
Comment (maikelmclauflin) [reply test/ledger/surveyor.integration.test.js]: sure. i can change it to `findOneSurveyor` or something

File: ledger/controllers/surveyor.js (hunk 2 above)
Comment (maikelmclauflin) [reply ledger/controllers/surveyor.js]: yeah that makes sense

File: test/ledger/surveyor.integration.test.js (hunk 4 above)
Comment (maikelmclauflin) [reply test/ledger/surveyor.integration.test.js]: fair enough

File: ledger/controllers/surveyor.js (hunk 1 above)
Comment (maikelmclauflin) [reply ledger/controllers/surveyor.js]: yes. can fix

File: package-lock.json [hunk 5]
```diff
@@ -115,19 +108,10 @@
...
-          "dev": true,
-          "requires": {
-            "minimist": "^1.2.0"
-          }
-        },
-        "minimist": {
-          "version": "1.2.0",
-          "resolved": "https://registry.npmjs.org/minimist/-/minimist-1.2.0.tgz",
-          "integrity": "sha1-o1AIsg9BOD7sH7kU9M1d95omQoQ=",
+        "ms": {
```
Comment (nvonpentz) [inline package-lock.json]: Looks like the most recent push also updated the node modules, is that intended?

Comment (maikelmclauflin) [reply package-lock.json]: yeah, looks like there was a conflict that wasn't showing on github from a recent merge

File: ledger/controllers/surveyor.js [hunk 6]
```diff
@@ -708,3 +739,15 @@ module.exports.initialize = async (debug, runtime) => {
     }
   }
 }
+
+function getVoteRate () {
+  return (runtime) => async (request, reply) => {
+    const surveyor = await server(request, reply, runtime)
+    console.log(surveyor)
```
Comment (nvonpentz) [inline ledger/controllers/surveyor.js]: Do you intend to commit this log line?

Comment (maikelmclauflin) [reply ledger/controllers/surveyor.js]: yeah, looks like i just forgot a step. was checking for where the altcurrency existed on the object. found it

**Pull request merged**